### PR TITLE
When TitleId is not set properly, there needs to be a very clear and …

### PR DESCRIPTION
**Pull request by JTonoJimenez**
…understandable warning about what happened

### CODE REVIEW

**Comment by JTonoJimenez** [comment]
retest

**Comment by pgilmorepf** [comment]
The only test that failed was Unity-Switch, which is unrelated to these changes.

File: targets/XPlatCppSdk/source/code/source/playfab/PlayFabWinHttpPlugin.cpp [hunk 1]
```diff
@@ -118,177 +118,183 @@ namespace PlayFab
...
-                                    }
-                                } // WinHttpQueryHeaders
-                            } // WinHttpReceiveResponse
-                        } // WinHttpSendRequest
-                    } // WinHttpOpenRequest
-                } // WinHttpConnect
-            } // WinHttpOpen
-        } // WinHttpCrackUrl
-
+		if (PlayFabSettings::verticalName.empty() && PlayFabSettings::titleId.empty())
```
Comment (pgilmorepf) [inline targets/XPlatCppSdk/source/code/source/playfab/PlayFabWinHttpPlugin.cpp]: Let's remove this condition:
PlayFabSettings::verticalName.empty() && 
The verticalName is always going to be empty right now (that feature isn't finished).

**Comment by pgilmorepf** [comment]
With the one change that I requested, this PR is approved.
The automated test will fail again, and I'll check it manually and then probably admin-override and land it.

**Comment by pgilmorepf** [comment]
retest

**Comment by pgilmorepf** [comment]
Stuff is powered up again so I want to check if it's working.

**Comment by pgilmorepf** [comment]
retest

**Comment by pgilmorepf** [comment]
retest

**Comment by JTonoJimenez** [comment]
retest

File: targets/XPlatCppSdk/source/code/source/playfab/PlayFabWinHttpPlugin.cpp [hunk 2]
```diff
@@ -118,177 +118,183 @@ namespace PlayFab
...
-                                    }
-                                } // WinHttpQueryHeaders
-                            } // WinHttpReceiveResponse
-                        } // WinHttpSendRequest
-                    } // WinHttpOpenRequest
-                } // WinHttpConnect
-            } // WinHttpOpen
-        } // WinHttpCrackUrl
-
+		if (PlayFabSettings::titleId.empty())
```
Comment (pgilmorepf) [inline targets/XPlatCppSdk/source/code/source/playfab/PlayFabWinHttpPlugin.cpp]: tabs -> 4 spaces please.

**Comment by Vicodex** [comment]
Please don't push any changes to this PR for a while, i will push change that will serve us as the foundation for a proper fix

**Comment by Vicodex** [comment]
retest

File: targets/XPlatCppSdk/templates/PlayFab_InstanceApi.cpp.ejs [hunk 3]
```diff
@@ -112,7 +113,10 @@ namespace PlayFab
         reqContainer->successCallback = std::shared_ptr<void>((callback == nullptr) ? nullptr : new ProcessApiCallback<<%- apiCall.result %>>(callback));
         reqContainer->errorCallback = errorCallback;
 
-        http.MakePostRequest(std::unique_ptr<CallRequestContainerBase>(static_cast<CallRequestContainerBase*>(reqContainer.release())));
+        if (PlayFabSettings::ValidateSettings("<%- apiCall.auth %>", authenticationContext, this->settings, *reqContainer))
```
Comment (pgilmorepf) [inline targets/XPlatCppSdk/templates/PlayFab_InstanceApi.cpp.ejs]: This is great.

**Pull request merged**